### PR TITLE
Adds template to urls

### LINKED ISSUE #64: Template variable in the link
I'm a Nextcloud admin.
I'd like to put links under quota that point to my billing system.
My billing system expects an email in the url for the bill.

## Implementation proposal

We create a function that allow Nextcloud admin to have special chars in the url, like say `{{ email }}`, and this would be replaced at runtime, by the actual email of the user.

so then, in the admin settings, I'd write:

```
https://example.org/buy_now/?email={{ email }}
```

And as a user, when I click on the link, I'd go to:
```
https://example.org/buy_now/?email=pierre@ozoux.net
```

What do you think about this feature?

### PR DESCRIPTION
**Pull request by pierreozoux**
closes #64

### CODE REVIEW

File: lib/Controller/APIController.php [hunk 1]
```diff
@@ -79,6 +91,10 @@ public function get() {
 				$site['icon'] = $this->url->linkToRouteAbsolute('external.icon.showIcon', ['icon' => 'external.svg']);
 			}
 
+			$site['url'] = str_replace('{email}', $email, $site['url']);
```
Comment (nickvergessen) [inline lib/Controller/APIController.php]: You can combine those 3 calls to one:
```php
$site['url'] = str_replace(['{email}', '{uid}', '{displayname}'], [$email, $uid, $displayname], $site['url']);
```

**Comment by pierreozoux** [comment]
@nickvergessen Thanks for the feedback, I corrected!

**Comment by pierreozoux** [comment]
Thanks a lot! I'm really excited!

**Comment by pierreozoux** [comment]
Could we make a release with that? Thanks!

**Comment by nickvergessen** [comment]
https://apps.nextcloud.com/apps/external/releases?platform=13#13

Sorry I was on vacation...

But this will only be available in the upcoming 13 release

**Comment by pierreozoux** [comment]
@nickvergessen no problems! thanks!

**Comment by dachinat** [comment]
Thanks for this feature. We have upgraded to 13.0.0 RC4, I have browsed apps/external/ directory and saw these changes, but I'm getting `<iframe id="ifm" src="https://lvh.me:3000/user/login?l=en&amp;uid={uid}&amp;email={email}&amp;displayname={displayname}"></iframe>` So, should this be working on 13.0.0 RC?

**Pull request merged**